app/src/test/java/com/github/se/cyrcle/model/user/UserRepositoryFirestoreTest.kt

```diff
@@ -55,92 +70,212 @@ class UserRepositoryFirestoreTest {
   }
 
   @Test
-  fun getAllUsers_returnsCorrectValues() {
-    // Create a TaskCompletionSource to manually complete the task
-    val taskCompletionSource = TaskCompletionSource<QuerySnapshot>()
+  fun getUidTest_throwsException_whenUidIsEmpty() {
+    `when`(mockFirebaseAuth.currentUser).thenReturn(null)
+    try {
+      userRepositoryFirestore.getUid()
+    } catch (e: Exception) {
+      assert(e.message == "User not signed in")
```

Comment:
If no exception is thrown, we don't enter the catch block. So we don't assert that the exception has been thrown

Reply:
You're right